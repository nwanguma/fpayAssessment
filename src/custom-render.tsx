import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const Wrapper: React.FC = ({ children }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
