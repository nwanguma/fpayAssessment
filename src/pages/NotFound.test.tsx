import { render, screen } from "../custom-render";

import NotFound from "./notFound";

describe("NotFound", () => {
  test("<NotFound /> matches snapshot", () => {
    const component = render(<NotFound />);

    expect(component.container).toMatchSnapshot();
  });

  test("It renders children correctly", () => {
    const App = render(<NotFound />);

    expect(screen.getByTestId("container")).toBeInTheDocument();
  });

  test("It renders the correct text", () => {
    render(<NotFound />);

    expect(screen.getByText("404")).toBeInTheDocument();
  });

  test("It renders the correct link", () => {
    render(<NotFound />);

    expect(screen.getByText("Home").closest("a")).toHaveAttribute("href", "/");
  });
});

export {};
