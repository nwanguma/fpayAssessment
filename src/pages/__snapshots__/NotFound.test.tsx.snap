// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound <NotFound /> matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="wrapper"
  >
    <header
      class="header--payout w-[100%] md:w-[90%] mx-auto flex items-center py-6 justify-between"
    >
      <a
        href="/"
      >
        <img
          alt="fliqpay"
          src="logo.svg"
        />
      </a>
    </header>
    <main
      class="container min-h-[80vh] min-w-full flex items-center"
    >
      <div
        class="container flex flex-col-reverse  md:flex-row justify-center md:justify-between items-center max-w-[400px] md:text-left md:max-w-[80%] mx-auto"
      >
        <div
          class="container mt-3 flex-item w-[90%] text-center md:text-left"
        >
          <div
            class="container"
            data-testid="container"
          >
            <h2
              class="text-[72px] leading-[90px] text-grey-large font-black"
            >
              404
            </h2>
            <h3
              class="mt-1 text-[24px] leading-[30px] text-blue-black font-bold"
            >
              Hmm, we couldnâ€™t find that page.
            </h3>
            <p
              class="mt-2 text-[16px] leading-[24px] text-grey-text"
            >
              No worries - we can still get you where you need to go. Our homepage is a good place to start.
            </p>
            <a
              class="flex items-center mx-auto md:-mx-0 justify-center leading-[40px] h-[40px] w-[172px] bg-blue-tab text-white-main btn mt-8"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
        <div
          class="container mx-auto md:mx-0 w-[250px] h-[150px] md:w-[350px] md:h-[250px]"
        >
          <img
            alt="error pic"
            class="w-[100%] h-[100%]"
            src="404.png"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
